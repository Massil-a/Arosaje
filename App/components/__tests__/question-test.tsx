import React from 'react';
import { render } from '@testing-library/react-native';
import Question from '../../app/optnav/question';

describe('Question', () => {
    it('renders correctly', () => {
        const { toJSON } = render(<Question />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('displays the correct number of questions', () => {
        const { getAllByText } = render(<Question />);
        const questions = getAllByText(/Comment arroser une plante ?|Quelle est la meilleure lumière pour les plantes ?/);
        expect(questions.length).toBe(2);
    });

    it('displays the correct answers', () => {
        const { getByText } = render(<Question />);
        expect(getByText('Arrosez les plantes lorsque le sol est sec.')).toBeTruthy();
        expect(getByText('La lumière indirecte est généralement la meilleure.')).toBeTruthy();
    });
});
