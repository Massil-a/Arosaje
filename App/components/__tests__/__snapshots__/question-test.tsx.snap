// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
<View
  style={
    {
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Comment arroser une plante ?
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
          "marginTop": 5,
        }
      }
    >
      Arrosez les plantes lorsque le sol est sec.
    </Text>
  </View>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Quelle est la meilleure lumière pour les plantes ?
    </Text>
    <Text
      style={
        {
          "fontSize": 14,
          "marginTop": 5,
        }
      }
    >
      La lumière indirecte est généralement la meilleure.
    </Text>
  </View>
</View>
`;
